### PR TITLE
Add RPC function and test

### PR DESCRIPTION
**Pull request by muheebyusufbaba1**
Closes https://github.com/keep-starknet-strange/rosettanet/issues/69

### CODE REVIEW

File: src/rpc/calls/getTransactionByBlockNumberAndIndex.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      message: 'Starknet RPC error',
+      data: 'Two parameters expected',
+    }
+  }
+
+  // Extract block number and index from the request parameters
+  const blockNumber = request.params[0] as number
+  const index = parseInt(request.params[1] as string, 16) // Convert index from hex to integer
+
+  // TODO: Validate block number
```
Comment (AryanGodara) [inline src/rpc/calls/getTransactionByBlockNumberAndIndex.ts:21]: Use this method
```
export function validateBlockNumber(value: any): boolean {
  if (typeof value === 'number') {
    return !isNaN(value) && value >= 0;
  }

  return false;
}
```

See #79 to see how it's implemented there :)

File: tests/rpc/calls/getTransactionByBlockNumberAndIndex.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    )
+
+    expect(starknetResult).toMatchObject({
+      code: 7979,
+      message: 'Starknet RPC error',
+      data: 'Empty result body',
+    })
+  })
+
+  it('Returns RPC error ', async () => {
```
Comment (AryanGodara) [inline tests/rpc/calls/getTransactionByBlockNumberAndIndex.test.ts:61]: The naming can be better. Currently it's difficult to guess "why" RPC error is returned
maybe try `invalid index param`?

File: tests/rpc/calls/getTransactionByBlockNumberAndIndex.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const starknetResult: RPCResponse = <RPCResponse>(
+      await getTransactionsByBlockNumberAndIndexHandler(request)
+    )
+    expect(starknetResult).toMatchObject({
+      code: 7979,
+      message: 'Starknet RPC error',
+      data: 'Transaction not found or index out of bounds',
+    })
+  })
+})
```
Comment (AryanGodara) [inline tests/rpc/calls/getTransactionByBlockNumberAndIndex.test.ts:81]: Add testcase for index out of bounds.

**Review by AryanGodara — CHANGES_REQUESTED** [review]
Just some small changes here and there

**Comment by AryanGodara** [comment]
> #69

Write `closes #69` so github can detect you're working on that issue ;)

**Comment by muheebyusufbaba1** [comment]
All comments raised have been fixed @AryanGodara

**Comment by AryanGodara** [comment]
> All comments raised have been fixed @AryanGodara

Awesome!

Also,
```txt
PARAMETERS:
QUANTITY|TAG - a block number, or the string "earliest", "latest" or "pending"
QUANTITY - the transaction index position.
```

Does the current implementation support me passing `earliest`, `latest`, and `pending`?

I think the validateBlockNumber function needs to be updated, and tests need to be added for these cases once you add this functionality

**Comment by ermvrs** [comment]
please resolve conflicts

**Pull request merged**